src/JavisNB.jl

```diff
@@ -0,0 +1,152 @@
+module JavisNB
+
+using Javis
+
+import Javis: AbstractObject, Video, StreamConfig
+
+import Interact
+import Interact: @map, @layout!, Widget
+
+export embed
+
+#################
+#### Pluto Viewer
+#################
+
+"""
+    _pluto_viewer(video::Video, frames::Int, actions::Vector)
+Creates an interactive viewer in a Pluto Notebook by storing all the frames in-memory
+```
+# In separate Pluto notebook cells
+using PlutoUI
+
+anim = render(
+    video;
+    pathname = "loading.gif",liveview=true
+);
+
+@bind x Slider(1:1:frames)
+
+anim[x]
+```
+"""
+function _pluto_viewer(video::Video, frames::Int, objects::Vector)
+    arr = collect(
+        Javis.get_javis_frame(video, objects, frame; layers = video.layers) for
+        frame in 1:frames
+    )
+    return arr
+end
+
+"""
+    PlutoViewer
+
+Wrapper to assist viewing rendered gifs as cell outputs of Pluto notebooks
+when `liveview = false` 
+"""
+struct PlutoViewer
+    filename::String
+end
+
+function Base.show(io::IO, ::MIME"image/png", v::PlutoViewer)
+    write(io, read(v.filename))
+end
+
+###################
+#### Jupyter Viewer
+################### 
+
+
+"""
+    _jupyter_viewer(video::Video, frames::Int, actions::Vector)
+Creates an interactive viewer in a Jupyter Notebook.
+"""
+function _jupyter_viewer(
+    video::Video,
+    frames::Int,
+    objects::Vector{AbstractObject},
+    framerate::Int,
+)
+    t = Interact.textbox(1:frames, value = 1, typ = Int)
+    f = Interact.slider(1:frames, label = "Frame", value = t)
+    obs = Interact.Observables.throttle(1 / framerate, f)
+    output = @map Javis.get_javis_frame(video, objects, &obs; layers = video.layers)
+    wdg = Widget(["f" => f, "t" => t], output = output)
+    @layout! wdg vbox(hbox(:f, :t), output)
+end
+
+
+##########
+#### embed
+##########
+
+"""
+    embed(pathname)
+
+This is the core function of `JavisNB` it allows to show the output of `render` 
+from [`Javis`](https://juliaanimators.github.io/Javis.jl/stable/) directly within 
+a notebook (IJulia or Pluto).
+
+To use it just wrap it around the `render``, am example within `Pluto`:
+
+```julia
+using Javis, JavisNB
+
+myVideo - Vide(...)
+
+# All the code here...
+
+embed(render(myVideo)...)
```

Comment:
I have to say that I am opposed to this syntax for embed as to me it looks odd from a user interface point of view.
An alternative I am wondering about is if it is possible to instead have a function called `notebook_render` that does not have the syntax for `liveview` but is exclusive to `JavisNB` and always assumes that whenever this function is called, to embed it in a given notebook.
This gets away from the splatting required (I think) and leaves one with the nice syntax of `notebook_render(myVideo)` as really, the function is just a special wrapper around the `Javis.render` command.
What do you think?

I am also a bit unclear as to why splatting is needed - is it to pass the values of `Object` and such?
I may just be missing something here because all it seems to me is to pass a `String` into `embed` at the moment.

Reply:
Hi @TheCedarPrince, so the splatting is gone, it was left in the docstring because I was thinking of making a different PR to add all the docs. There was however a small issue with livewview but it is now gone and the docstring for embed now look also better.

Let me know what you think

Reply:
Ah this makes a lot of sense because I was having issues when trying to run this from within a Pluto notebook!
Thanks!